Source/Urho3D/AngelScript/APITemplates.h

```diff
@@ -232,19 +232,20 @@ template <class BaseType, class DerivedType> void RegisterSubclass(asIScriptEngi
 {
     if (!strcmp(baseClassName, derivedClassName))
         return;
+    String strCast = String(baseClassName) +
+                            ((engine->GetTypeInfoByName(baseClassName)->GetFlags() & asOBJ_NOCOUNT) ? "@" : "@+") +
+                            " opImplCast()";
+    engine->RegisterObjectMethod(derivedClassName, strCast.CString(), AS_FUNCTION_OBJLAST((RefCast<DerivedType, BaseType>)), AS_CALL_CDECL_OBJLAST);
 
-    String declReturnBase(String(baseClassName) + "@+ opImplCast()");
-    engine->RegisterObjectMethod(derivedClassName, declReturnBase.CString(), AS_FUNCTION_OBJLAST((RefCast<DerivedType, BaseType>)), AS_CALL_CDECL_OBJLAST);
-
-    String declReturnBaseConst("const " + String(baseClassName) + "@+ opImplCast() const");
+    String declReturnBaseConst("const " + strCast + " const");
     engine->RegisterObjectMethod(derivedClassName, declReturnBaseConst.CString(), AS_FUNCTION_OBJLAST((RefCast<DerivedType, BaseType>)), AS_CALL_CDECL_OBJLAST);
 
-    //String declReturnDerived(String(derivedClassName) + "@+ opCast()");
-    String declReturnDerived(String(derivedClassName) + "@+ opImplCast()");
-    engine->RegisterObjectMethod(baseClassName, declReturnDerived.CString(), AS_FUNCTION_OBJLAST((RefCast<BaseType, DerivedType>)), AS_CALL_CDECL_OBJLAST);
+    strCast = String(derivedClassName) +
+              ((engine->GetTypeInfoByName(derivedClassName)->GetFlags() & asOBJ_NOCOUNT) ? "@" : "@+") +
+              " opImplCast()";
+    engine->RegisterObjectMethod(baseClassName, strCast.CString(), AS_FUNCTION_OBJLAST((RefCast<BaseType, DerivedType>)), AS_CALL_CDECL_OBJLAST);
 
-    //String declReturnDerivedConst("const " + String(derivedClassName) + "@+ opCast() const");

```

Comment:
I was planning to do an explicit conversion from base to derived as recommended here https://www.angelcode.com/angelscript/sdk/docs/manual/doc_adv_class_hierarchy.html (but this require fix many scripts)

Reply:
May be some build option to backward compatibility